appcues/src/main/java/com/appcues/analytics/ExperienceLifecycleEvent.kt

```diff
@@ -16,7 +20,12 @@ internal sealed class ExperienceLifecycleEvent(
     data class StepInteraction(
         override val experience: Experience,
         val stepIndex: Int,
-    ) : ExperienceLifecycleEvent(experience, AnalyticsEvent.ExperienceStepInteraction)
+        val interactionType: InteractionType,
+    ) : ExperienceLifecycleEvent(experience, AnalyticsEvent.ExperienceStepInteraction) {
+        enum class InteractionType {
+            FORM_SUBMITTED
+        }
```

Comment:
right now this is the only type, but leaving this extensible for the future to try to match the spirit of the event spec